src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Serialization.cs

```diff
@@ -181,26 +181,49 @@ public void WriteTo(ObjectWriter writer)
                     }
                 }
             }
+
+            // sortedNodes is an array of Node instances which is often sorted by Node.Name by the caller. This method
+            // produces a sequence of spans within sortedNodes for Node instances that all have the same Name, allowing
+            // serialization to record the string once followed by the remaining properties for the nodes in the group.
+            static IEnumerable<ReadOnlyMemory<Node>> GroupByName(ImmutableArray<Node> sortedNodes)
+            {
+                if (sortedNodes.IsEmpty)
+                    yield break;
+
+                var startIndex = 0;
+                var currentName = sortedNodes[0].Name;
+                for (var i = 0; i < sortedNodes.Length; i++)
+                {
+                    var node = sortedNodes[i];
+                    if (node.Name != currentName)
+                    {
+                        yield return sortedNodes.AsMemory()[startIndex..i];
```

Comment:
nit: should we hoist out the AsMemory view higher up?  i think it would help make this, and the below call nicer.

Reply:
Now moved up